protocol/0087-VAMM-automated_market_maker.md

```diff
@@ -0,0 +1,177 @@
+
+```
+{
+  commitment,
+  market,
+  slippage_tolerance_percentage,
+  concentrated_liquidity_params: {
+    base_price,
+    lower_price,
+    upper_price,
+    margin_ratio_at_bounds,
+  }
+}
+```
+
+### Concentrated Liquidity
+
+The `Concentrated Liquidity` AMM is a market maker utilising a Uniswap v3-style pricing curve for managing price based upon current market price. This allows for the market maker to automatically provide a pricing curve for any prices within some configurable range, alongside offering the capability to control risk by only trading within certain price bounds and out to known position limits.
+
+The concentrated liquidity market maker consists of two liquidity curves of prices joined at a given `base price`, an `upper` consisting of the prices above this price point and a `lower` for prices below it. At prices below the `base price` the market maker will be in a long position, and at prices above this `base price` the market maker will be in a short position. This is configured through a number of parameters:
+
+- **Base Price**: The base price is the central price for the market maker. When trading at this level the market maker will have a position of `0`. Volumes for prices above this level will be taken from the `upper` curve and volumes for prices below will be taken from the `lower` curve.
+- **Upper Price**: The maximum price bound for market making. Prices between the `base price` and this price will have volume placed, with no orders above this price. This is optional and if not supplied no volume will be placed above `base price`. At these prices the market maker will always be short
+- **Lower Price**: The minimum price bound for market making. Prices between the `base price` and this will have volume placed, with no orders below this price. This is optional and if not supplied no volume will be placed below `base price`. At these prices the market maker will always be long
+- **Commitment**: This is the initial volume of funds to transfer into the sub account for use in market making. If this amount is not currently available in the main account's general account the transaction will fail.
+- **Margin Ratio at Bounds**: The exact volume scaling is defined by the position at the upper and lower prices. To determine this we must compare the commitment with what leverage that might allow at the price bounds. One way to do this is to assume we will use the value at which `commitment == initial margin` for the position at that price, however users may wish to take a more conservative approach. Using this parameter allows them to set a value such that `position = commitment / margin ratio`, however with the restriction that commitment must still be `>= initial margin`. This parameter should be optional.
```

Comment:
Should we have different ratios at the upper and lower bounds?

Reply:
For consistency let's also write `margin_ratio_at_bounds` here.

Reply:
Done